framework/src/kernels/Kernel.C

```diff
@@ -80,6 +80,25 @@ Kernel::computeJacobian()
 
   ke += _local_ke;
 
+  if (_calculate_nonlocal_variable_jacobian)
+  {
+    _nonlocal_ke.resize(ke.m(), _subproblem.es().n_dofs());
+    _nonlocal_ke.zero();
+    for (_i = 0; _i < _test.size(); _i++)
```

Comment:
_subproblem.es().n_dofs() or _sys.n_dofs() gives the total number of dofs in the system. How do I get the total dofs for a single variable? Also any suggestion about what would be a good place to modify the sparse matrix jacobian?